WalletWasabi/Io/IoManager.cs

```diff
@@ -79,6 +79,41 @@ protected static StreamReader OpenText(string filePath, int bufferSize = Constan
 			return new StreamReader(fs, Encoding.ASCII, detectEncodingFromByteOrderMarks: true, bufferSize: bufferSize, leaveOpen: false);
 		}
 
+		public async Task WriteAllLinesAsync(IEnumerable<string> lines, CancellationToken cancellationToken = default, bool dismissNullOrEmptyContent = true)
+		{
+			if (ShouldReturn(lines, dismissNullOrEmptyContent))
+			{
+				return;
+			}
+
+			IoHelpers.EnsureContainingDirectoryExists(FilePath);
+
+			await File.WriteAllLinesAsync(FilePath, lines, cancellationToken);
+		}
+
+		public async Task AppendAllLinesAsync(IEnumerable<string> lines, CancellationToken cancellationToken = default)
+		{
+			if (lines is null || !lines.Any())
+			{
+				return;
+			}
+
+			await File.AppendAllLinesAsync(FilePath, lines, cancellationToken);
+		}
+
+		protected static bool ShouldReturn(IEnumerable<string> lines, bool dismissNullOrEmptyContent)
```

Comment:
I would rename this to `IsUpdateRequired()`. `ShouldReturn` describe how it is used but not he logic inside.

Reply:
I removed it, because we never actually used that parameter that made this complex.